app/javascript/packs/stocks.js

```diff
@@ -0,0 +1,65 @@
+// import "select/dist/select.js"
+import 'select2'
+import 'select2/dist/css/select2.css'
+import '@ttskch/select2-bootstrap4-theme/dist/select2-bootstrap4.min.css'
+
+$(function () {
+  console.log("stocks")
+  let action_name = $('body').data('action')
+  console.log(action_name);
+
+  const csrfToken = document.querySelector('[name="csrf-token"]').getAttribute('content');
+
+  switch (action_name) {
+    case "new":
+      $(".selectfa").select2({
+        ajax: {
+          url: '/stocks/search_rawmaterial',
+          datatype: 'json',
+          delay: 500,
+          data: function (params) {
+            return { q: params.term }
+          },
+          processResults: function (data, params) {
+            return {
+              results: $.map(data, function (obj) {
+                return { id: obj.id, text: obj.name };
+              })
+            };
+          }
+        },
+        theme: 'bootstrap4',
+      })
+
+      $(".select2-results__option").on('click', function () {
+        console.log($(this))
+        console.log("aaa")
+      })
```

Comment:
公式の書き方だと、これでイベント登録できそうです。
https://select2.org/programmatic-control/events

```suggestion
      }).on('select2:open', function () {
        console.log($(this))
        console.log("aaa")
      })
```